### ISSUE TITLE
empty_parentheses_with_trailing_closure false positive

### ISSUE THREAD

**Issue by peacemoon** [realm/SwiftLint#1160]
We have a class with a init function that take a closure as parameter

```
class Promise {
    public init(_ f: (something) -> Void))
}
```
when I call it
```
Promise() { 
    // doing something here
}
```

the rule  empty_parentheses_with_trailing_closure will give a warning even if the call is totally ok.

**Comment by marcelofabri** [comment]
Actually, the rule enforces exactly this. You should use:

```swift
Promise { _ in
    print("foo")
}
```

(Or disable the rule if you don't agree with this style)

**Comment by peacemoon** [comment]
ok, thank you.
